### ISSUE TITLE
Typescript support

### ISSUE THREAD

**Issue by timkinnane** [typicode/json-server#833]
Would be great to see some definitions for this package on https://github.com/DefinitelyTyped/DefinitelyTyped/

**Comment by MickL** [comment]
Not just definitions, the whole project is not working with ts files.

works:
json-server --watch db.js

does not work:
json-server --watch db.ts

**Comment by MickL** [comment]
Not just definitions, this module is not working with TypeScript at all.

Works:
json-server --watch db.js

Does not work:
json-server --watch db.ts

**Comment by amivit** [comment]
I'm really wanting this too. It makes so much sense defining interfaces for all the API contracts you'd like to configure. Especially if the frontend will be Angular you can reuse the same interfaces. If we simply point json-server to the compiled typescript would it work? In which case how to setup an empty nodejs project with typescript

**Comment by jeremyben** [comment]
Just did it: https://github.com/DefinitelyTyped/DefinitelyTyped/pull/31107

`npm install -D @types/json-server`
`yarn add -D @types/json-server`

🙂

**Comment by MickL** [comment]
Is this working now?

json-server --watch db.ts

**Comment by jeremyben** [comment]
No, just that the programmatic use of json-server (https://github.com/typicode/json-server#module) is now properly typed, as OP asked.

What you want is doable with that programmatic use and ts-node (https://github.com/TypeStrong/ts-node).

**Comment by MickL** [comment]
That works, unfortunately without `--watch` but we can use nodemon for that.

Anyway I think the important part of using TypeScript is to have a ts-file instead of a json-file for the db so the responses are typed, but json-server simply parses the file provided by `jsonServer.router()` as json no matter what.

**Comment by moepkid** [comment]
> `ts-node json-server.ts`

Can you ellaborta how you got the export working? I am getting the error: 'unexpected token {' ont he import db line. I am developing on angular 7 and running 

kill-port 3000 && concurrently --kill-others "nodemon --exec ts-node json-server.ts" "ng serve\""

**Comment by crascher** [comment]
to fix the "unexpected token" error see: https://stackoverflow.com/a/51390319/4237304
nevertheless would be nice to have support for a db.ts ouf the box

**Comment by andresberrios** [comment]
I just submitted https://github.com/typicode/json-server/pull/994 to allow the CLI to use files with extensions other than `.js` to allow it to load the DB from a TypeScript file. The only thing you would need to do once this is merged and published is registering the `ts-node` loader directly in the node executable by doing:

`NODE_OPTIONS="--require ts-node/register" json-server db.ts`

**Comment by SerkanSipahi** [comment]
hi, i had the same issue: i found this and its works:
`nodemon --ext ts --watch src --exec './node_modules/.bin/ts-node' src/www.ts`
see resource: https://github.com/jamg44/NodeTyped/blob/master/package.json#L10

**Comment by gekkedev** [comment]
I tried to make it even shorter:
```
yarn run ts-node `yarn bin`/json-server
```
You can still configure the json-server via CLI here but in many cases you just want to go with the defaults.

**Comment by MickL** [comment]
Two years later I come here again with the exact same question and I thank my past self for writing the documentation :) 

Anyhow too bad this project does not seem to be maintained anymore @typicode . Over 500 issues. Open PR for TypeScript integration... Maybe someone wants to make a fork?

**Comment by SalahAdDin** [comment]
Four years passed man!

> Two years later I come here again with the exact same question and I thank my past self for writing the documentation :)
> 
> Anyhow too bad this project does not seem to be maintained anymore @typicode . Over 500 issues. Open PR for TypeScript integration... Maybe someone wants to make a fork?
